### ISSUE TITLE
Extending with custom media types on contract loading

### ISSUE THREAD

**Issue by ppatierno** [eclipse-vertx/vertx-openapi#95]
We have been using the vertx-web-openapi since years within our Strimzi HTTP bridge component in order to create routes from our own OpenAPI v3 specification JSON file.
We are now in the process to try Vert.x 5.x which is removing such component and replacing it with the vertx-web-openapi-router which uses the vertx-openapi.
The issue we are facing is that in our OpenAPI v3 spec we have custom media types, so something like `application/vnd.kafka.v2+json`.
In the end this is just JSON (so compatible with the `application/json` one) but we preferred to have our own custom type.
The issue is that when using the `OpenAPIContract.from(vertx, "openapi.json")` method we get the following exception:

```
io.vertx.openapi.contract.OpenAPIContractException: The passed OpenAPI contract contains a feature that is not supported: Operation createConsumer defines a request body with an unsupported media type. Supported: application/json, application/json; charset=utf-8, multipart/form-data, application/hal json
```

The exception itself is pretty clear even looking at the code and how the vertx-openapi works by supporting a limited list of media-types.
My question is what's the way to overcome this problem by adding a custom media type and avoid such an error? I can't find any information about how much this is pluggable without changing the openapi.json file and moving to use `application/json`.

**Comment by pk-work** [comment]
Hi @ppatierno thanks for using Vert.x OpenAPI! Yes, at the moment this is a limitation which affects a lot of users. Solving this issue is one of my highest priorities. I need to figure out a good way to support custom implementations of `ContentAnalyser`. Unfortunately since a few weeks / months my time I can spent on this is very limited.

If you have time to work on a fix for this I would appreciate it.  If so, we should briefly discuss the planned idea. It would be bad to spend implementation effort on a solution that may not fit into the library.

Otherwise you just have to wait a little bit longer :)

**Comment by pk-work** [comment]
duplicate https://github.com/eclipse-vertx/vertx-openapi/issues/18

**Comment by tsegismont** [comment]
@pk-work we discussed off-list the feasibility of configuration synonyms for content types. Any chance you could provide a PR for this in the short term?

**Comment by ppatierno** [comment]
That would be great because we are already testing Vert.x 5 RC4 in our main operator but cannot proceed with the Strimzi HTTP bridge because of this issue. I also see that Vert.x 5 GA is expected in the first quarter of 2025 so within 2 months I guess.

**Comment by pk-work** [comment]
@tsegismont @ppatierno, yes I can provide a short term fix as discussed. But the short term fix requires to hard code the synonyms. Please write down the synonyms you need. Or is `application/vnd.kafka.v2+json` the only one?

Due to the fact that it is just a synonym, the payload of `application/vnd.kafka.v2+json` is pure JSON, isn't it?

**Comment by ppatierno** [comment]
> But the short term fix requires to hard code the synonyms. Please write down the synonyms you need.

What does "hard code" exactly mean? Isn't it possible to extend the openapi library to be a configurable thing? I mean you are putting some synonyms related to a specific application (the Strimzi HTTP bridge) within a general purpose library. Is that a good approach?

> Or is application/vnd.kafka.v2+json the only one?

No. We have 4 of them I have to double check.

> the payload of application/vnd.kafka.v2+json is pure JSON, isn't it?

Yes it's still JSON.

**Comment by pk-work** [comment]
"Hard code" means I will add the synonyms here [1].

> Isn't it possible to extend the openapi library to be a configurable thing?

Sure it is! And this is also my goal! But we talked about short term fix I can provide now. As I said at the moment I'm very limited. If you want to provide a configurable solution, lets have a sync and then you can provide one.

[1] https://github.com/eclipse-vertx/vertx-openapi/blob/da9eef32f56a1e057172ce1147b571e786198498/src/main/java/io/vertx/openapi/validation/analyser/ContentAnalyser.java#L66

**Comment by ppatierno** [comment]
@pk-work @tsegismont I had a look at the code and of course the best solution would be adding pluggable content analyser but I have opened a draft PR https://github.com/eclipse-vertx/vertx-openapi/pull/97 to where I am avoiding to hard code specific content type but a more generic vendor specific JSON as per the defintion of "vendor specific" within RFC 6838. The PR also adds support for `text/plain` which was missing as well and we need. Tests are not there and I am going to write them if you think that the proposed PR is a good way to go. Let me know please.

**Comment by pk-work** [comment]
Looks good to me, just one comment because of code duplication. Please add missing tests, then we can merge it.
